File: Problems/968_Binary_Tree_Cameras.cpp
#include<iostream>

// Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
};

class Solution {
public:
    int camera_count = 0;
    int minCameraCover(TreeNode* root) {
        if (traverse(root)){
            return camera_count;
        }else{
            return camera_count+1;
        }
    }
    // None: 0, Covered: 1, Camera: 2
    int traverse(TreeNode* node){
        if (! node){
            return 1;
        }
        int l = traverse(node->left);
        int r = traverse(node->right);
        if (l==0 || r==0){
            camera_count++;
            return 2;
        }
        if (l==2 || r==2){
            return 1;
        }
        return 0;
    }
};

File: Problems/968_Binary_Tree_Cameras.py
from typing import Optional

# Definition for a binary tree node.
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def __init__(self) -> None:
        self.camera_count = 0
    def minCameraCover(self, root: Optional[TreeNode]) -> int:

        # None: 0, Covered: 1, Camera: 2
        def traverse(node):
            if not node:
                return 1
            l = traverse(node.left)
            r = traverse(node.right)
            if l == 0 or r == 0:
                self.camera_count += 1
                return 2
            if l == 2 or r == 2:
                return 1
            return 0
        
        if traverse(root):
            return self.camera_count
        else:
            return self.camera_count+1